components/org.wso2.carbon.identity.query.saml/src/main/java/org/wso2/carbon/identity/query/saml/handler/SAMLAssertionFinderImpl.java

```diff
@@ -123,44 +119,39 @@ public Assertion findByID(String id) throws IdentitySAML2QueryException {
      */
     public List<Assertion> findBySubjectAndSessionIndex(String subject, String sessionIndex)
             throws IdentitySAML2QueryException {
-        List<Assertion> assertions = new ArrayList<Assertion>();
+
+        List<Assertion> assertions;
         Connection connection = null;
         PreparedStatement preparedStatement = null;
         ResultSet resultSet = null;
         try {
-            connection = JDBCPersistenceManager.getInstance().getDBConnection();
-            String selectBySubjectQuery = "SELECT SAML2_ASSERTION FROM IDN_SAML2_ASSERTION_STORE WHERE SAML2_SUBJECT =? " +
-                    "AND SAML2_SESSION_INDEX=? ";
-            preparedStatement = connection.prepareStatement(selectBySubjectQuery);
+            connection = JDBCPersistenceManager.getInstance().getDBConnection();;
+            preparedStatement = connection.prepareStatement(DBUtil.SELECT_BY_SESSION_QUERY);
             preparedStatement.setString(1, subject);
             preparedStatement.setString(2, sessionIndex);
             resultSet = preparedStatement.executeQuery();
-            int index = 1;
-            while (resultSet.next()) {
-                String assertionString = resultSet.getString(index);
-                assertions.add((Assertion) SAMLQueryRequestUtil.unmarshall(assertionString));
+            assertions = extractAssertions(resultSet);
+            if(log.isDebugEnabled() && assertions.size() > 0) {
```

Comment:
Fix code formatting issues in all places.

Reply:
Add checkstyle plugin: https://medium.com/@jayanga/how-to-configure-checkstyle-and-findbugs-plugins-to-intellij-idea-for-wso2-products-c5f4bbe9673a